primitives/block/src/equivocation_proof.rs

```diff
@@ -405,13 +419,27 @@ impl DoubleProposalProof {
 
         if self.header1.block_number != self.header2.block_number
             || self.header1.round != self.header2.round
```

Comment:
Is this check still correct or should it go away?

Reply:
I would say this was not correct before and it should be removed. The round in the header only denotes when a proposal was created, not when it was proposed. Thus it should not be checked here.